frontend/test/views/AccountView.spec.js

```diff
@@ -83,34 +88,85 @@ describe('AccountView.vue', () => {
   });
 
   it('should not toggle the secret value on click again after it has been toggled', async () => {
     const wrapper = getMountedComponent();
 
-    let sectionBox = wrapper.findComponent('[data-test=credentials]');
+    let credentialsBox = wrapper.findComponent('[data-test=credentials]');
+
+    await credentialsBox.vm.$emit('display-secret');
+    await wrapper.vm.$nextTick();
+
+    credentialsBox = wrapper.findComponent('[data-test=credentials]');
+    expect(credentialsBox.props('clientSecret')).to.equal(wrapper.vm.clientSecret);
+    await credentialsBox.vm.$emit('display-secret');
+    await wrapper.vm.$nextTick();
+
+    credentialsBox = wrapper.findComponent('[data-test=credentials]');
+    expect(credentialsBox.props('clientSecret')).to.equal(wrapper.vm.clientSecret);
+  });
+
+  it('should call onGenerateSecret method and update clientSecret', async () => {
+    getAPICredentialsStub.returns({client_secret: 'newSecret'});
+
+    const wrapper = getMountedComponent();
+    const credentialsBox = wrapper.findComponent('[data-test=credentials]');
+
+    await credentialsBox.vm.$emit('generateSecret');
+    await wrapper.vm.$nextTick();
+
+    expect(generateSecretStub.called).to.be.true;
+    expect(getAPICredentialsStub.called).to.be.true;
+    expect(wrapper.vm.clientSecret).to.equal('newSecret');
+  });
+
+  it('should update showSecretValue to true when updateSecretValue is called', () => {
+    const wrapper = getMountedComponent();
+    wrapper.vm.updateSecretValue();
+    expect(wrapper.vm.showSecretValue).to.be.true;
+  });
+
+  it('should update clientSecret on multiple generateSecret calls', async () => {
+    getAPICredentialsStub.onFirstCall().returns({client_secret: 'newSecret1'});
+    getAPICredentialsStub.onSecondCall().returns({client_secret: 'newSecret2'});
 
-    await sectionBox.trigger('click');
+    const wrapper = getMountedComponent();
+    const credentialsBox = wrapper.findComponent('[data-test=credentials]');
+
+    await credentialsBox.vm.$emit('generateSecret');
     await wrapper.vm.$nextTick();
+    expect(generateSecretStub.calledOnce).to.be.true;
+    expect(getAPICredentialsStub.calledOnce).to.be.true;
+    expect(wrapper.vm.clientSecret).to.equal('newSecret1');
 
-    sectionBox = wrapper.findComponent('[data-test=credentials]');
-    expect(sectionBox.props('content')[1].value).toEqual([wrapper.vm.user.clientSecret]);
-    await sectionBox.trigger('click');
+    await credentialsBox.vm.$emit('generateSecret');
     await wrapper.vm.$nextTick();
+    expect(generateSecretStub.calledTwice).to.be.true;
+    expect(getAPICredentialsStub.calledTwice).to.be.true;
+    expect(wrapper.vm.clientSecret).to.equal('newSecret2');
+  });
 
-    sectionBox = wrapper.findComponent('[data-test=credentials]');
-    expect(sectionBox.props('content')[1].value).toEqual([wrapper.vm.user.clientSecret]);
+  it('should return clientSecret when showSecretValue is true', () => {
```

Comment:
As mentioned, this and the test on line 163 are implementation detail https://test-utils.vuejs.org/guide/essentials/easy-to-test.html#do-not-test-implementation-details unit tests.  How do these values affect the components? Prop value changes? Rendering? Think about the observable changes that are happening.